### PR TITLE
[FuseReshape] Reject fusion when block extent may exceed real dimension extent

### LINKED ISSUE #7464: [Pytorch upstream] `tl.make_tensor_descriptor` 3D non-K-major with **constexpr** padded strides still produces incorrect `tl.dot` results (gap in #7165 fix)
### Describe the bug

Follow-up to #7030 / #7165. The original #7030 reproducer (strides passed as
runtime kernel args) now passes. But the same miscompile still occurs when the
descriptor **strides are `tl.constexpr`** (which is what PyTorch Inductor emits)
and padded (`stride_K > innermost dim`), in the 3D non-K-major descriptor +
`reshape` + `tl.dot(a.T, b)` path.

### Reproducer

```python
import torch
import triton
import triton.language as tl


def alloc_fn(size, align, stream):
    return torch.empty(size, dtype=torch.int8, device="xpu")
triton.set_allocator(alloc_fn)


@triton.jit
def kern(a_ptr, b_ptr, c_ptr):
    # shapes/strides are compile-time constants, exactly like Inductor emits.
    G: tl.constexpr = 5
    M: tl.constexpr = 3
    N: tl.constexpr = 7
    K: tl.constexpr = 13
    A_STRIDE_G: tl.constexpr = 104
    A_STRIDE_K: tl.constexpr = 8   # padded: stride_K(8) > M(3)
    A_STRIDE_M: tl.constexpr = 1
    B_STRIDE_G: tl.constexpr = 104
    B_STRIDE_K: tl.constexpr = 8   # padded: stride_K(8) > N(7)
    B_STRIDE_N: tl.constexpr = 1
    C_STRIDE_G: tl.constexpr = 21
    C_STRIDE_M: tl.constexpr = 7
    C_STRIDE_N: tl.constexpr = 1
    BLOCK_M: tl.constexpr = 16
    BLOCK_N: tl.constexpr = 64
    BLOCK_K: tl.constexpr = 64

    a_desc = tl.make_tensor_descriptor(
        a_ptr, shape=[G, K, M],
        strides=[A_STRIDE_G, A_STRIDE_K, A_STRIDE_M],
        block_shape=[1, BLOCK_K, BLOCK_M])
    b_desc = tl.make_tensor_descriptor(
        b_ptr, shape=[G, K, N],
        strides=[B_STRIDE_G, B_STRIDE_K, B_STRIDE_N],
        block_shape=[1, BLOCK_K, BLOCK_N])

    for g in tl.range(G):
        acc = tl.zeros((BLOCK_M, BLOCK_N), dtype=tl.float32)
        n_full = K // BLOCK_K
        for ki in range(n_full):
            k = ki * BLOCK_K
            a = a_desc.load([g, k, 0]).reshape(BLOCK_K, BLOCK_M)
            b = b_desc.load([g, k, 0]).reshape(BLOCK_K, BLOCK_N)
            acc += tl.dot(a.T, b)
        if K % BLOCK_K != 0:
            k = n_full * BLOCK_K
            a = a_desc.load([g, k, 0]).reshape(BLOCK_K, BLOCK_M)
            b = b_desc.load([g, k, 0]).reshape(BLOCK_K, BLOCK_N)
            acc += tl.dot(a.T, b)
        c = acc.to(tl.bfloat16)
        offs_m = tl.arange(0, BLOCK_M)
        offs_n = tl.arange(0, BLOCK_N)
        c_ptrs = (c_ptr + g * C_STRIDE_G
                  + offs_m[:, None] * C_STRIDE_M + offs_n[None, :] * C_STRIDE_N)
        tl.store(c_ptrs, c, mask=(offs_m[:, None] < M) & (offs_n[None, :] < N))


def main():
    torch.manual_seed(42)
    G, M, N, K = 5, 3, 7, 13
    A = torch.randn(G, K, 8, dtype=torch.bfloat16, device="xpu").transpose(-2, -1)[:, :M, :]
    mat_b = torch.randn(G, K, 8, dtype=torch.bfloat16, device="xpu").transpose(-2, -1)[:, :N, :].transpose(-2, -1)
    assert A.stride() == (104, 1, 8), A.stride()
    assert mat_b.stride() == (104, 8, 1), mat_b.stride()
    ref = torch.bmm(A, mat_b)
    C = torch.zeros(G, M, N, dtype=torch.bfloat16, device="xpu")
    kern[(1,)](A, mat_b, C)
    d = (ref - C).abs().max().item()
    print(f"max diff {d:.3f}  {'PASS' if d < 0.1 else '*** FAIL ***'}")


if __name__ == "__main__":
    main()
```

Output:

```
max diff 16.000  *** FAIL ***
```

Note: making the strides runtime kernel args (instead of `constexpr`) makes it
pass. Pointer-based loads are always correct.

### Environment details

- Device: Intel Data Center GPU Max 1100 (PVC), `gpu_subslice_count=56`
- Triton (triton-xpu): 3.8.0, built from source, commit
  `9810c442983223928bd1cc96fa858938a2b50fa0` (already contains the #7030 fixes
  #7114 / #7165)
- PyTorch: built from source (main)
- OS: Linux

### PR DESCRIPTION
**Pull request by anmyachev**

### CODE REVIEW

**Comment by anmyachev** [comment]
I would like to have it for the release as well. Please review.

**Review by chengjunlu — APPROVED** [review]

**Comment by chengjunlu** [comment]
Changes looks good to me.

The conservertive optimization is acceptable since we are going to formally support the >2 rank tensor desc prefetching and load in plan.

After that, this pass is not need.
